src/common/ipc/ipc.ts

```diff
@@ -48,51 +45,52 @@ function getSubFrames(): ClusterFrameInfo[] {
   return Array.from(clusterFrameMap.values());
 }
 
-export function broadcastMessage(channel: string, ...args: any[]) {
-  const subFramesP = ipcRenderer
-    ? requestMain(subFramesChannel)
-    : Promise.resolve(getSubFrames());
-
-  subFramesP
-    .then(subFrames => {
-      const views: undefined | ReturnType<typeof webContents.getAllWebContents> | ReturnType<typeof remote.webContents.getAllWebContents> = (webContents || electronRemote?.webContents)?.getAllWebContents();
-
-      if (!views || !Array.isArray(views) || views.length === 0) return;
-      args = args.map(sanitizePayload);
-
-      ipcRenderer?.send(channel, ...args);
-      ipcMain?.emit(channel, ...args);
-
-      for (const view of views) {
-        let viewType = "unknown";
-
-        // There will be a uncaught exception if the view is destroyed.
-        try {
-          viewType = view.getType();
-        } catch {
-          // We can ignore the view destroyed exception as viewType is only used for logging.
-        }
-
-        // Send message to views.
-        try {
-          logger.silly(`[IPC]: broadcasting "${channel}" to ${viewType}=${view.id}`, { args });
-          view.send(channel, ...args);
-        } catch (error) {
-          logger.error(`[IPC]: failed to send IPC message "${channel}" to view "${viewType}=${view.id}"`, { error: String(error) });
-        }
-
-        // Send message to subFrames of views.
-        for (const frameInfo of subFrames) {
-          logger.silly(`[IPC]: broadcasting "${channel}" to subframe "frameInfo.processId"=${frameInfo.processId} "frameInfo.frameId"=${frameInfo.frameId}`, { args });
-
-          try {
-            view.sendToFrame([frameInfo.processId, frameInfo.frameId], channel, ...args);
-          } catch (error) {
-            logger.error(`[IPC]: failed to send IPC message "${channel}" to view "${viewType}=${view.id}"'s subframe "frameInfo.processId"=${frameInfo.processId} "frameInfo.frameId"=${frameInfo.frameId}`, { error: String(error) });
-          }
-        }
+export async function broadcastMessage(channel: string, ...args: any[]): Promise<void> {
+  if (ipcRenderer) {
+    return requestMain(broadcastMainChannel, ...args);
+  }
+
+  if (!webContents) {
+    return;
+  }
+
+  const subFrames = getSubFrames();
+  const views = webContents.getAllWebContents();
+
+  if (!views || !Array.isArray(views) || views.length === 0) return;
+  args = args.map(sanitizePayload);
+
+  ipcMain?.emit(channel, ...args);
+
+  for (const view of views) {
+    let viewType = "unknown";
+
+    // There will be a uncaught exception if the view is destroyed.
+    try {
+      viewType = view.getType();
```

Comment:
Maybe better to check `view.isDestroyed()` instead of try-catch or it might catch something else?